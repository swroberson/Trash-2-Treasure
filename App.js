import 'react-native-gesture-handler';
import { staticBlock } from "@babel/types";
import { Button, Text, TextInput, View, StyleSheet, SafeAreaView } from "react-native";
import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createDrawerNavigator } from '@react-navigation/drawer';
import StackNavigation from "./src/navigation/StackNavigation"

const Drawer = createDrawerNavigator();

function App() {
  return (
      <NavigationContainer> 
        <StackNavigation/>
      </NavigationContainer>
  );
};

function HomeScreen () {
  return (
      <Drawer.Navigator initialRouteName="Map">
        <Drawer.Screen name="Map" component={MapScreen} />
        <Drawer.Screen name="View Map" component={MapScreen} />
      </Drawer.Navigator>
  );
};

function MapScreen() {
  return (
  <Button
    title="View Map"
  />
  );
};

export default App;