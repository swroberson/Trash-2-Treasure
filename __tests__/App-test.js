import "react-native";
import React from "react";
import App from "../App";

import { fireEvent, render, waitFor } from "@testing-library/react-native";

it("Renders Message 1", async () => {
  const { getByTestId, getByText, queryByTestId, toJSON } = render(<App />);

  const button = getByText("Run Test 1");
  fireEvent.press(button);

  await waitFor(() => expect(queryByTestId("printed-message1")).toBeTruthy());

  expect(getByTestId("printed-message1").props.children).toBe("1");
  expect(toJSON()).toMatchSnapshot();
}),

it("Renders Message 2", async () => {
  const { getByTestId, getByText, queryByTestId, toJSON } = render(<App />);

  const button = getByText("Run Test 2");
  fireEvent.press(button);

  await waitFor(() => expect(queryByTestId("printed-message2")).toBeTruthy());

  expect(getByTestId("printed-message2").props.children).toBe("2");
  expect(toJSON()).toMatchSnapshot();
})

it("Renders Message 3", async () => {
  const { getByTestId, getByText, queryByTestId, toJSON } = render(<App />);

  const button = getByText("Run Test 3");
  fireEvent.press(button);

  await waitFor(() => expect(queryByTestId("printed-message3")).toBeTruthy());

  expect(getByTestId("printed-message3").props.children).toBe("3");
  expect(toJSON()).toMatchSnapshot();
});