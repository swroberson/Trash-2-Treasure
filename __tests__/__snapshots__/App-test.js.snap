// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Message 1 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Array [
        undefined,
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Run Test 1
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Run Test 2
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Run Test 3
        </Text>
      </View>
    </View>
  </View>
  <View>
    <Text
      style={
        Object {
          "fontFamily": "Arial",
          "fontSize": 38,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
      testID="printed-message1"
    >
      1
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`Renders Message 2 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Array [
        undefined,
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Run Test 1
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Run Test 2
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Run Test 3
        </Text>
      </View>
    </View>
  </View>
  <View>
    <Text
      style={
        Object {
          "fontFamily": "Arial",
          "fontSize": 38,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
      testID="printed-message2"
    >
      2
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`Renders Message 3 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Array [
        undefined,
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Run Test 1
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Run Test 2
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Run Test 3
        </Text>
      </View>
    </View>
  </View>
  <View>
    <Text
      style={
        Object {
          "fontFamily": "Arial",
          "fontSize": 38,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
      testID="printed-message3"
    >
      3
    </Text>
  </View>
</RCTSafeAreaView>
`;
